### PR TITLE
Improve error msg on invalid onConflict option

### PR DESCRIPTION
**Pull request by ebrattli**

### CODE REVIEW

File: src/test/scala/com/cognite/spark/datasource/AssetsRelationTest.scala [hunk 1]
```diff
@@ -148,6 +149,31 @@ class AssetsRelationTest extends FlatSpec with Matchers with SparkTest {
       .save
   }
 
+  it should "Should give good error on invalid onconflict options" taggedAs WriteTest in {
```
Comment (Tapped) [inline src/test/scala/com/cognite/spark/datasource/AssetsRelationTest.scala]: remove `Should` from the string literal

File: src/main/scala/com/cognite/spark/datasource/DefaultSource.scala [hunk 2]
```diff
@@ -24,6 +24,8 @@ case class RelationConfig(
 object OnConflict extends Enumeration {
   type Mode = Value
   val ABORT, UPDATE, UPSERT, DELETE = Value
+
+  def withNameOpt(s: String): Option[Value] = values.find(_.toString.toLowerCase == s.toLowerCase())
```
Comment (Tapped) [inline src/main/scala/com/cognite/spark/datasource/DefaultSource.scala]: Can't we use
https://www.scala-lang.org/api/current/scala/Enumeration.html#withName(s:String):Enumeration.this.Value?

Or is that method not case insensitive?

**Comment by wjoel** [comment]
Small detail, but it adds up over time: please capitalize the first word in the commit message. We generally try to follow the guidelines in https://chris.beams.io/posts/git-commit/

File: src/main/scala/com/cognite/spark/datasource/DefaultSource.scala (hunk 2 above)
Comment (ebrattli) [reply src/main/scala/com/cognite/spark/datasource/DefaultSource.scala]: The problem with that method is that it throws the exception referenced in the related issue. The new function returns an Option and allows us to throw a more specific error message.

Comment (Tapped) [reply src/main/scala/com/cognite/spark/datasource/DefaultSource.scala]: Oh, I see then it makes sense. And it's also nicer than relying on exception for code flow, as when doing `Try(OnConflict.withName(parameters.getOrElse("onconflict", "ABORT").toUpperCase())).toOption`.

**Pull request merged**